packages/openapi-to-graphql/src/resolver_builder.ts

```diff
@@ -559,92 +574,54 @@ export function getResolver<TSource, TContext, TArgs>({
      * GraphQL produces sanitized payload names, so we have to sanitize before
      * lookup here
      */
-    resolveData.usedPayload = undefined
+    let payload
     if (typeof payloadName === 'string') {
-      // The option genericPayloadArgName will change the payload name to "requestBody"
+      // The option `genericPayloadArgName` will change the payload name to "requestBody"
       const sanePayloadName = data.options.genericPayloadArgName
         ? 'requestBody'
         : Oas3Tools.sanitize(payloadName, Oas3Tools.CaseStyle.camelCase)
 
-      let rawPayload
       if (operation.payloadContentType === 'application/json') {
-        rawPayload = JSON.stringify(
+        payload = JSON.stringify(
           Oas3Tools.desanitizeObjectKeys(args[sanePayloadName], data.saneMap)
         )
       } else if (
         operation.payloadContentType === 'application/x-www-form-urlencoded'
       ) {
-        rawPayload = formurlencoded(
+        payload = formurlencoded(
           Oas3Tools.desanitizeObjectKeys(args[sanePayloadName], data.saneMap)
         )
       } else {
         // Payload is not an object
-        rawPayload = args[sanePayloadName]
+        payload = args[sanePayloadName]
       }
-      options.body = rawPayload
-      resolveData.usedPayload = rawPayload
     }
 
-    /**
-     * Pass on OpenAPI-to-GraphQL options
-     */
-    if (typeof data.options === 'object') {
-      // Headers:
-      if (typeof data.options.headers === 'object') {
-        Object.assign(options.headers, data.options.headers)
-      } else if (typeof data.options.headers === 'function') {
-        const headers = data.options.headers(method, path, title, {
-          source,
-          args,
-          context,
-          info
-        })
-
-        if (typeof headers === 'object') {
-          Object.assign(options.headers, headers)
-        }
-      }
-
-      // Query string:
-      if (typeof data.options.qs === 'object') {
-        Object.assign(options.qs, data.options.qs)
-      }
+    let options: NodeRequest.OptionsWithUrl = {
+      method: operation.method,
+      url,
+      headers,
+      qs,
+      body: payload,
+      jar: cookieJar // For authentication cookies
     }
 
-    // Get authentication headers and query parameters
-    if (
-      source &&
-      typeof source === 'object' &&
-      typeof source[OPENAPI_TO_GRAPHQL] === 'object'
-    ) {
-      const { authHeaders, authQs, authCookie } = getAuthOptions(
-        operation,
-        source[OPENAPI_TO_GRAPHQL],
-        data
-      )
-
-      // ...and pass them to the options
-      Object.assign(options.headers, authHeaders)
-      Object.assign(options.qs, authQs)
-
-      // Add authentication cookie if created
-      if (authCookie !== null) {
-        const j = NodeRequest.jar()
-        j.setCookie(authCookie, options.url)
-        options.jar = j
-      }
-    }
+    // The options `requestOptions` will finalize the options
+    if (requestOptions) {
```

Comment:
`requestOptions` has been moved to the very bottom, right before the call. This allows `requestOptions` to have the final say on how the request should be made.